Some examples don't run with newer versions of NPM

The cause of this issue is `npm install` failing for versions of npm@v7. See output below.

See the Peer Dependencies section here: https://blog.npmjs.org/post/626173315965468672/npm-v7-series-beta-release-and-semver-major

This causes npm install to fail in certain cases. For example:

### Potential Fixes

- Replace all uses of `npm install` with `npm install --legacy-peer-deps` across the different builders/architects.
- Update [the versions for peer dependencies in the package.json](https://github.com/facebookresearch/Mephisto/blob/master/packages/mephisto-task/package.json#L39-L40) in the mephisto-task pkg by prefixing with ^ to be more lenient

### Output

```console
$ cd mephisto/examples/simple_static_task && python static_test_script.py
[2021-01-27 04:42:55,653][mephisto.operations.operator][INFO] - Creating a task run under task name: html-static-task-example

added 123 packages, and audited 124 packages in 8s

3 high severity vulnerabilities

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.
[2021-01-27 04:43:03,347][sh.command][INFO] - <Command '/bin/rm -rf /mephisto/data/data/runs/NO_PROJECT/1/build/router', pid 19>: process started
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: mephisto-task-compiler@1.0.0
npm ERR! Found: react@16.14.0
npm ERR! node_modules/react
npm ERR!   react@"^16.5.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"16.13.1" from mephisto-task@1.0.14
npm ERR! node_modules/mephisto-task
npm ERR!   mephisto-task@"^1.0.14" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /root/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-01-27T04_43_04_367Z-debug.log
[2021-01-27 04:43:04,386][mephisto.operations.operator][ERROR] - Encountered error while launching run, shutting down
Traceback (most recent call last):
  File "/mephisto/mephisto/operations/operator.py", line 210, in validate_and_run_config_or_die
    built_dir = architect.prepare()
  File "/mephisto/mephisto/abstractions/architects/local_architect.py", line 132, in prepare
    self.server_dir = build_router(self.build_dir, self.task_run)
  File "/mephisto/mephisto/abstractions/architects/router/build_router.py", line 89, in build_router
    task_builder.build_in_dir(local_server_directory_path)
  File "/mephisto/mephisto/abstractions/blueprints/static_html_task/static_html_task_builder.py", line 62, in build_in_dir
    self.rebuild_core()
  File "/mephisto/mephisto/abstractions/blueprints/static_html_task/static_html_task_builder.py", line 45, in rebuild_core
    raise Exception(
Exception: please make sure npm is installed, otherwise view the above error for more info.
[2021-01-27 04:43:04,389][mephisto.operations.operator][ERROR] - Could not shut down architect: shutdown called before deploy
Traceback (most recent call last):
  File "/mephisto/mephisto/operations/operator.py", line 210, in validate_and_run_config_or_die
    built_dir = architect.prepare()
  File "/mephisto/mephisto/abstractions/architects/local_architect.py", line 132, in prepare
    self.server_dir = build_router(self.build_dir, self.task_run)
  File "/mephisto/mephisto/abstractions/architects/router/build_router.py", line 89, in build_router
    task_builder.build_in_dir(local_server_directory_path)
  File "/mephisto/mephisto/abstractions/blueprints/static_html_task/static_html_task_builder.py", line 62, in build_in_dir
    self.rebuild_core()
  File "/mephisto/mephisto/abstractions/blueprints/static_html_task/static_html_task_builder.py", line 45, in rebuild_core
    raise Exception(
Exception: please make sure npm is installed, otherwise view the above error for more info.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/mephisto/mephisto/operations/operator.py", line 262, in validate_and_run_config_or_die
    architect.shutdown()
  File "/mephisto/mephisto/abstractions/architects/local_architect.py", line 175, in shutdown
    assert self.running_dir is not None, "shutdown called before deploy"
AssertionError: shutdown called before deploy
[2021-01-27 04:43:04,389][mephisto.operations.operator][ERROR] - Ran into error while launching run:
Traceback (most recent call last):
  File "/mephisto/mephisto/operations/operator.py", line 412, in validate_and_run_config
    return self.validate_and_run_config_or_die(
  File "/mephisto/mephisto/operations/operator.py", line 268, in validate_and_run_config_or_die
    raise e
  File "/mephisto/mephisto/operations/operator.py", line 210, in validate_and_run_config_or_die
    built_dir = architect.prepare()
  File "/mephisto/mephisto/abstractions/architects/local_architect.py", line 132, in prepare
    self.server_dir = build_router(self.build_dir, self.task_run)
  File "/mephisto/mephisto/abstractions/architects/router/build_router.py", line 89, in build_router
    task_builder.build_in_dir(local_server_directory_path)
  File "/mephisto/mephisto/abstractions/blueprints/static_html_task/static_html_task_builder.py", line 62, in build_in_dir
    self.rebuild_core()
  File "/mephisto/mephisto/abstractions/blueprints/static_html_task/static_html_task_builder.py", line 45, in rebuild_core
    raise Exception(
Exception: please make sure npm is installed, otherwise view the above error for more info.
[2021-01-27 04:43:04,390][mephisto.operations.operator][INFO] - operator shutting down
```

--- a/mephisto/abstractions/blueprints/static_html_task/source/package.json
+++ b/mephisto/abstractions/blueprints/static_html_task/source/package.json
@@ -13,9 +13,9 @@
     "bootstrap": "^4.3.1",
     "mephisto-task": "^1.0.14",
     "rc-slider": "^8.6.3",
-    "react": "^16.5.2",
+    "react": "16.13.1",
     "react-bootstrap": "^0.32.4",
-    "react-dom": "^16.5.2",
+    "react-dom": "16.13.1",
     "react-table": "^6.8.6"
   },
   "devDependencies": {

